waku/v2/node/jsonrpc/filter_api.nim

```diff
@@ -0,0 +1,44 @@
+import
+  json_rpc/rpcserver,
+  eth/[common, rlp, keys, p2p],
+  ../../waku_types,
+  ../wakunode2
+
+proc installFilterApiHandlers*(node: WakuNode, rpcsrv: RpcServer) =
+  const futTimeout = 5.seconds
+
+  ## Filter API version 1 definitions
+  
+  rpcsrv.rpc("post_waku_v2_filter_v1_subscription") do(contentFilters: seq[ContentFilter], topic: Option[string]) -> bool:
+    ## Subscribes a node to a list of content filters
+    debug "post_waku_v2_filter_v1_subscription"
+
+    proc filterHandler(msg: WakuMessage) {.gcsafe, closure.} =
+      debug "WakuMessage received", msg=msg, topic=topic
+      # @TODO handle message
+
+    # Construct a filter request
+    # @TODO use default PubSub topic if undefined
+    let fReq = if topic.isSome: FilterRequest(topic: topic.get, contentFilters: contentFilters, subscribe: true) else: FilterRequest(contentFilters: contentFilters, subscribe: true)
+    
+    if (await node.subscribe(fReq, filterHandler).withTimeout(futTimeout)):
+      # Successfully subscribed to all content filters
+      return true
+    else:
+      # Failed to subscribe to one or more content filters
+      raise newException(ValueError, "Failed to subscribe to contentFilters " & repr(fReq))
```

Comment:
https://github.com/status-im/nim-waku/issues/289 at top of file I believe

Reply:
Thanks! Will address in next PR.